### ISSUE TITLE
Proposition on `to_be_*` (and similar) syntax.

### ISSUE THREAD

**Issue by beregond** [dgilland/verify#7]
Instead of writing `expect(some_call()).Equals(3)` you could type `exepect(some_call()).to_be_equal(3)`. In other words it's another step of what was introduced in #2, but now it's more readable, you can make assertions (almost) in english.

You could still type

```
Expect(True).Truthy().Boolean().Equal(1)
```

but you could also switch to something like:

```
expect(True).to_be_truthy().to_be_boolean().to_be_equal(1)
```

Of course some validators would require different prefix (`is_even`, `is_odd`, `in_between` etc).

That would improve IMHO the best part of Verify - readability in tests.

If you are interested, we could cooperate to provide this feature.

**Comment by dgilland** [comment]
I'm not opposed to having function aliases that are more readable/closer to english phrases.

I would prefer it if the prefix is consistent for all functions so that one doesn't have to remember what the exceptions are, but I believe that won't be possible (without some unnatural phrases).

I believe `Even`, `Odd`, and `Between` sound fine as `to_be_even`, `to_be_odd`, and `to_be_between`.

Some examples that don't work well, though:
- `[Not]Contains[Only]` as `to_be_[not_]contains[_only]`: Maybe `to_[not_]contain[_only]` would be an okay compromise in keeping with at least having a `to_` prefix.
- `Length` as `to_be_length`: This is not too bad but might sound better as `to_have_length`.

Other unnatural cases are the `Not*` functions as `to_be_not_*` doesn't sound right (e.g. `expect(...).to_be_not_false()`). So might be better as `to_not_be_*` (e.g. `expect(...).to_not_be_false()`).

However, another approach to using `expect.to_be/have_*` style would be to use `expect.is/has_` or if that's too unnatural (e.g. `expect(True).is_truthy()` vs `expect(True).to_be_truthy()`) use `ensure` instead of `expect` (e.g. `ensure(True).is_truthy()`).

Or we could do all of it and have `ensure` be an alias of `expect`, all assertions have aliases for `to_be_*` (taking into account exceptions), and all assertions have aliases for `is_*` (taking into account exceptions). So you could do:

``` python
expect(True).Truthy().Boolean().Equal(1)
expect(True).to_be_truthy().to_be_boolean().to_be_equal(1)
expect(True).is_truthy().is_boolean().is_equal(1)

ensure(True).Truthy().Boolean().Equal(1)
ensure(True).to_be_truthy().to_be_boolean().to_be_equal(1)
ensure(True).is_truthy().is_boolean().is_equal(1)
```

So to summarize the options:
1. Implement `to_be_*` prefixes only.
2. Implement `is_*` prefixes only.
3. Implement `is_*` prefixes and `ensure` alias.
4. Implement `to_be_*` prefixes and `is_*` prefixes.
5. Implement `to_be_*` prefixes, `is_*` prefixes, and `ensure` alias.

**Comment by beregond** [comment]
Great, I think I'll go with option 5. :)

**Comment by dgilland** [comment]
This has been released with [v1.1.0](https://pypi.python.org/pypi/verify/1.1.0).

**Comment by beregond** [comment]
@dgilland thanks!
